src/main/java/org/pf4j/update/FileVerifiers.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2018 Decebal Suiu
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.pf4j.update;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Collection of standard file verifiers.
+ * Other implementations could e.g. be Antivirus checks, scanning for forbidden jar files,
+ * checking digital signatures etc.
+ */
+public class FileVerifiers {
+  /**
+   * All default verifiers
+   */
+  public static final List<FileVerifier> ALL_DEFAULT_FILE_VERIFIERS = Arrays.asList(
+      new FileVerifiers.BasicVerifier(),
+      new Sha512SumVerifier());
+
+  /**
+   * Verifies that the SHA512 checksum of a downloaded file equals the checksum given in
+   * the plugins.json descriptor. This helps validate that the file downloaded is exactly
+   * the same as intended. Especially useful when dealing with meta repositories pointing
+   * to S3 or other 3rd party download locations that could have been tampered with.
+   */
+  public static class Sha512SumVerifier implements FileVerifier {
+    private static final Logger log = LoggerFactory.getLogger(SimpleFileDownloader.class);
```

Comment:
It's something wrong with indentation (two spaces instead of four).

Reply:
Will fix